examples/tictactoe/src/model.rs

```diff
@@ -0,0 +1,174 @@
+use alloc::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+use valerie::store::{Mutator, Relation};
+use valerie::{remote, singleton};
+
+#[derive(Copy, Clone, PartialEq, Debug, Deserialize)]
+pub enum SquareMark {
+    Empty,
+    X,
+    O,
+}
+
+impl Default for SquareMark {
+    fn default() -> Self {
+        Self::Empty
+    }
+}
+
+#[derive(Debug)]
+pub enum NextPlayerChange {
+    Start,
+    Next,
+}
+
+impl Mutator<SquareMark> for NextPlayerChange {
+    fn mutate(&self, v: &SquareMark) -> SquareMark {
+        use SquareMark::*;
+        match self {
+            Self::Start => X,
+            Self::Next => match v {
+                X => O,
+                O => X,
+                Empty => Empty,
+            },
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug, Serialize, Deserialize)]
+pub struct SquareID(u8);
+
+#[derive(Copy, Clone, Deserialize)]
+pub struct Square {
+    id: SquareID,
+    pub mark: SquareMark,
+}
+
+impl Square {
+    pub fn new(id: SquareID) -> Self {
+        Self {
+            id,
+            mark: SquareMark::default(),
+        }
+    }
+}
+
+impl Default for Square {
+    fn default() -> Self {
+        Square {
+            id: SquareID(0),
+            mark: SquareMark::default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum SquareChange {
+    Mark(SquareMark),
+}
+
+impl Mutator<Arc<Square>> for SquareChange {
+    fn mutate(&self, v: &Arc<Square>) -> Arc<Square> {
+        let mut v: Square = Square::clone(v);
+        match self {
+            Self::Mark(mark) => v.mark = *mark,
+        }
+        Arc::new(v)
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct Board {
+    pub squares: [SquareID; 9],
+}
+
+impl Default for Board {
+    fn default() -> Self {
+        let mut squares = [SquareID(0); 9];
+        for i in 0usize..9 {
+            let square = Square::new(SquareID(i as u8));
+            Square::new(square.id);
```

Comment:
This line here is questionable: the ::get which occurs later could just use the ::default and fetch the actual value..